### PR TITLE
Update go version to 1.19

### PR DESCRIPTION
**Pull request by kisunji**
Upgrading the go version to 1.19

Highly recommended to review by commit.

Includes:
- General CI/CD version bumps
- Changelog
- go1.19-style gofmt

Note:
The linter `unparam` has not been active in go1.18 due to a bug. This resulted in several features developed recently (peering) to fail the linter in minor ways (unused return variable). I've added TODO's in those files to clean them up

### CODE REVIEW

**Review by DanStough — CHANGES_REQUESTED** [review]
I was taking a look at https://github.com/hashicorp/consul/pull/12808/files for reference, and I see a few more places to change:
* .github/workflows/build.yaml
* build-support/docker/Build-Go.dockerfile
* (new) test/integration/consul-container/go.mod

TBH I'm not really sure if we do changelogs for these changes, so maybe it's not necessary

**Comment by kisunji** [comment]
> I was taking a look at https://github.com/hashicorp/consul/pull/12808/files for reference, and I see a few more places to change:
> 
> * .github/workflows/build.yaml
> * build-support/docker/Build-Go.dockerfile
> * (new) test/integration/consul-container/go.mod
> 
> TBH I'm not really sure if we do changelogs for these changes, so maybe it's not necessary

Thanks for catching these! I grepped for 1.18 but those didn't come up for some reason 🤔

**Comment by david-yu** [comment]
We typically do include changelog details as its helpful to know when go was updated https://github.com/hashicorp/consul/blob/main/CHANGELOG.md#1120-april-20-2022

**Comment by kisunji** [comment]
1.19 formatting 😭

File: .circleci/bash_env.sh [hunk 1]
```diff
@@ -6,5 +6,5 @@ export GIT_DIRTY=$(test -n "`git status --porcelain`" && echo "+CHANGES" || true
 export GIT_IMPORT=github.com/hashicorp/consul/version
 # we're using this for build date because it's stable across platform builds
 # the env -i and -noprofile are used to ensure we don't try to recursively call this profile when starting bash
-export GIT_DATE=$(env -i /bin/bash --noprofile -norc ${CIRCLE_WORKING_DIRECTORY}/build-support/scripts/build-date.sh)
```
Comment (kisunji) [inline .circleci/bash_env.sh:9]: this was not expanding correctly: https://discuss.circleci.com/t/circle-working-directory-doesnt-expand/17007

this caused all CI runs to output:
```
~/project/build-support/scripts/build-date.sh: No such file or directory
```

File: agent/consul/acl_endpoint.go [hunk 2]
```diff
@@ -108,19 +108,18 @@ type ACL struct {
...
-//    bootstrap "index" into a special file on disk to override the mechanism preventing
-//    a second token bootstrap. The index will be retrieved by a API call to /v1/acl/bootstrap
-//    When already bootstrapped this API will return the reset index necessary within
-//    the error response. Once set in the file, the bootstrap API can be used again to
-//    get a new token.
-//
-// Q: Why is the reset index not in the config?
-// A: We want to be able to remove the reset index once we have used it. This prevents
-//    accidentally allowing bootstrapping yet again after a snapshot restore.
+//		Q: What is the bootstrap reset index?
```
Comment (rboyer) [inline agent/consul/acl_endpoint.go:111]: Was this double indent of the first Q/A intentional?

Comment (kisunji) [reply agent/consul/acl_endpoint.go:111]: Nope, this is a victim of tabs vs spaces

File: proto/pbsubscribe/subscribe.pb.go [hunk 3]
```diff
@@ -320,6 +320,7 @@ type SubscribeRequest struct {
 	// receive events (e.g. health events for a particular service).
 	//
 	// Types that are assignable to Subject:
+	//
```
Comment (rboyer) [inline proto/pbsubscribe/subscribe.pb.go:323]: Did you have to edit this, or did you run `make proto` and it did the right thing?

Comment (kisunji) [reply proto/pbsubscribe/subscribe.pb.go:323]: `make proto` formatted everything

**Review by rboyer — APPROVED** [review]

**Pull request merged**